src/gpu/intel/gemm/jit/gen_kernel.cpp

```diff
@@ -487,10 +487,12 @@ gen_nocopy_desc_t::select_kernel(compute::gpu_arch_t arch, int stepping,
         problem_.batch = BatchMode::Strided;
         problem_.batchDims = batch_dims;
     }
-    if (a_quant.zp_ndims >= 0) problem_.aOffset = ABOffset::Calc;
-    if (b_quant.zp_ndims >= 0) problem_.bOffset = ABOffset::Calc;
-    problem_.aoPtrDims = a_quant.zp_ndims;
-    problem_.boPtrDims = b_quant.zp_ndims;
+    if (a_quant.zp_ndims >= 0 || a_quant.zp_hostscalar)
+        problem_.aOffset = ABOffset::Calc;
+    if (b_quant.zp_ndims >= 0 || a_quant.zp_hostscalar)
+        problem_.bOffset = ABOffset::Calc;
+    problem_.aoPtrDims = a_quant.zp_hostscalar ? -1 : a_quant.zp_ndims;
+    problem_.boPtrDims = b_quant.zp_hostscalar ? -1 : b_quant.zp_ndims;
```

Comment:
I recommend changing `quant_entry_ndims` to return -1 for host-side scalar, then adding the variable `has_zp` to `quant_params_t`, which would make this logic more straightforward.

```
problem_.aOffset = a_quant.has_zp ? ABOffset::Calc : ABOffset::None;
problem_.bOffset = b_quant.has_zp ? ABOffset::Calc : ABOffset::None;
problem_.aoPtrDims = a_quant.zp_ndims;
problem_.boPtrDims = b_quant.zp_ndims;
```

Reply:
I see. At some point we should refactor to store all quantization information directly in `quant_params` when calling `init_attr` and have these accessors/queries use this. Then adding `has_zp` to the struct could simplify these functions as well, to `bool with_a_zero_points() const { return a_quant.has_zp; }`. But I'm ok with a later refactor, since that affects other cases unrelated to this PR.